### PR TITLE
Fixed bug #7270

### PR DESCRIPTION
**Pull request by jlorente**
Fixed bug #7271. Classes that call '_::populateRecord()' after retrieving data from ddbb call now the method on the class of the model returned by '_::instantiate()'. Tests for mysql, elasticsearch and mongodb attached.

### CODE REVIEW

**Comment by Faryshta** [comment]
maybe you got the issue number wrong

**Comment by qiangxue** [comment]
Please see the comment https://github.com/yiisoft/yii2/issues/7271#issuecomment-74321411

File: extensions/elasticsearch/ActiveQuery.php [hunk 1]
```diff
@@ -215,7 +215,8 @@ public function one($db = null)
             /* @var $class ActiveRecord */
             $class = $this->modelClass;
             $model = $class::instantiate($result);
-            $class::populateRecord($model, $result);
+            $_class = get_class($model);
```
Comment (qiangxue) [inline extensions/elasticsearch/ActiveQuery.php]: Variable name needs to be changed. We never use leading underscore in local variable names.

File: tests/unit/data/ar/Animal.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * @author Jose Lorente <jose.lorente.martin@gmail.com>
+ * @property integer $id
+ * @property string $type
+ */
+class Animal extends ActiveRecord {
+
+    public $does;
+
+    public static function tableName() {
```
Comment (qiangxue) [inline tests/unit/data/ar/Animal.php]: The curly bracket needs to go to the next line, according to our coding style. Please fix them all in this PR.

File: tests/unit/extensions/elasticsearch/ActiveRecordTest.php [hunk 3]
```diff
@@ -443,24 +429,22 @@ public function testBooleanAttribute()
...
+    public function testScriptFields() {
         $orderItems = OrderItem::find()->fields([
-            'quantity',
-            'subtotal',
-            'total' => [
-                'script' => "doc['quantity'].value * doc['subtotal'].value",
-                'lang' => 'groovy',
-            ]
-        ])->all();
+                    'quantity',
```
Comment (qiangxue) [inline tests/unit/extensions/elasticsearch/ActiveRecordTest.php]: Please keep the original indentation (similar in the following code). In general, try not to fix the coding style of the original code while you are fixing a bug. You may submit a separate PR is coding style needs to be adjusted massively. Thanks.

**Comment by jlorente** [comment]
Ok, I will fix the problems and submit them. Sorry about the coding style, it is my first contribution to Yii.

**Comment by cebe** [comment]
@jlorente you can push a new commit to the same branch to update the pull request, no need to close it.
